include/CMakeLists.txt

```diff
@@ -1,11 +1,20 @@
+set(Boost_USE_STATIC_LIBS OFF)
+find_package(Boost REQUIRED system)
 
 file(GLOB HDR_ROOT "*.hpp")
 file(GLOB HDR_MQTT "mqtt/*.hpp")
 
-add_library(mqtt_cpp_iface INTERFACE)
+add_library(${PROJECT_NAME} INTERFACE)
 
-target_include_directories(mqtt_cpp_iface INTERFACE ${HDR_ROOT} ${HDR_MQTT})
+target_link_libraries(${PROJECT_NAME} INTERFACE ${Boost_SYSTEM_LIBRARIES} -lpthread)
```

Comment:
I think that `${Boost_SYSTEM_LIBRARIES}` isn't defined.
According to the document https://cmake.org/cmake/help/v3.8/module/FindBoost.html

```
Boost_<C>_LIBRARY      - Libraries to link for component <C> (may include
                         target_link_libraries debug/optimized keywords)
```

So you need to use `${Boost_SYSTEM_LIBRARY}`


`-lpthread` is platform dependent.

See
https://github.com/redboltz/mqtt_cpp/blob/master/CMakeLists.txt#L42
https://github.com/redboltz/mqtt_cpp/blob/master/test/CMakeLists.txt#L36